packages/react-bindings/src/contexts/StreamVideoContext.tsx

```diff
@@ -30,6 +36,26 @@ export const StreamVideo = ({
   language,
   translationsOverrides,
 }: PropsWithChildren<StreamVideoProps>) => {
+  const prevClient = useRef<StreamVideoClient | undefined>(undefined);
+
+  useEffect(() => {
+    if (prevClient.current) {
+      prevClient.current
+        .disconnectUser()
+        .catch((error) => console.error(`Failed to disconnect`, error));
+    }
+
+    if (client.user) {
+      client
+        .connectUser(client.user, client.token)
+        .catch((error) =>
+          console.error('Failed to establish connection', error),
+        );
+    }
+
+    prevClient.current = client;
```

Comment:
We should probably return a cleanup function that would disconnect the client in case this component would be unmounted, but the customer's app keeps running.

Reply:
Done